app/src/main/kotlin/io/homeassistant/companion/android/settings/SettingsFragment.kt

```diff
@@ -359,6 +359,24 @@ class SettingsFragment(
                 return@setOnPreferenceClickListener true
             }
         }
+
+        findPreference<SwitchPreference>("enable_ha_launcher")?.let { switchPreference ->
+            switchPreference.setOnPreferenceClickListener {
+                findPreference<Preference>("set_launcher_app")?.isVisible = switchPreference.isChecked
+                true
+            }
+        }
+
+        findPreference<Preference>("set_launcher_app")?.let {
+            it.isVisible = findPreference<SwitchPreference>("enable_ha_launcher")?.isChecked ?: false
+            it.summary = getString(commonR.string.default_launcher_prompt_def, getDefaultLauncherInfo())
+            it.setOnPreferenceClickListener {
+                val intent = Intent(Settings.ACTION_HOME_SETTINGS)
+                    .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                startActivity(intent)
+                true
+            }
+        }
```

Comment:
The class is already quite big (same for the function itself) I would be in favor of grouping this two findPref into one private method.